src/SignalR/clients/cpp/test/signalrclienttests/connection_impl_tests.cpp

```diff
@@ -111,6 +111,28 @@ TEST(connection_impl_start, connection_state_is_disconnected_when_connection_can
     ASSERT_EQ(connection->get_connection_state(), connection_state::disconnected);
 }
 
+TEST(connection_impl_start, throws_for_invalid_uri)
+{
+    std::shared_ptr<log_writer> writer(std::make_shared<memory_log_writer>());
+
+    auto websocket_client = create_test_websocket_client(
+        /* receive function */ []() { return pplx::task_from_result(std::string("{ }\x1e")); });
+
+    auto connection = connection_impl::create(_XPLATSTR("/bad_uri?a=b"), trace_level::errors, writer, create_test_web_request_factory(), std::make_unique<test_transport_factory>(websocket_client));
+
+    try
+    {
+        connection->start().get();
+        ASSERT_TRUE(false);
+    }
+    catch (...)
+    {
+        // We shouldn't check the exact exception as it would be specific to the http library being used
```

Comment:
What is the exception with the default http library? Does it make it clear the URI is invalid?

Reply:
"provided uri is invalid: :1	 ÃƒÂ¤ bad_uri&a=b"